# ProyectoDS_ParteI_CruzRazoAdolfo
## Análisis de padecimiento de la diabetes en EEUU 

## ABSTRACT

### Resumen

Este proyecto se centra en analizar la prevalencia y las causas de la diabetes en Estados Unidos, utilizando un conjunto de datos equilibrado basado en la encuesta BRFSS 2015 del CDC, disponible en Kaggle. La diabetes, una enfermedad crónica que impacta significativamente la salud pública y la economía, afecta a millones de personas en el país. En la primera fase, se busca analizar y visualizar hallazgos clave en los datos de más de 70,000 participantes, lo que permitirá en fases posteriores desarrollar modelos predictivos para mejorar el diagnóstico y tratamiento de la diabetes.

Dataset:
    Teboul, A. (2019). Diabetes Health Indicators Dataset. Kaggle. https://www.kaggle.com/datasets/alexteboul/diabetes-health-indicators-dataset



### Hipótesis de interés:
#### Basándonos en un análisis previo del conjunto de datos, planteamos las siguientes hipótesis:
- La prevalencia de diabetes varía según el nivel educativo.
- Factores como la obesidad y la hipertensión aumentan el riesgo de diabetes.
- La actividad física y la dieta influyen en el desarrollo de diabetes.
- El consumo del alcohol y tabaco influyen en el padecimiento de la diabetes.
- La edad y el género influyen en el padecimiento de la diabetes.

### Objetivos:
- Probar las hipótesis mediante análisis estadísticos y visualizaciones.
- Identificar hallazgos importantes en los datos mediante visualizaciones.
- Mostrar la información de forma fácil y sencilla de comprender para la población  .

### Impacto:
#### Los resultados permitirán:
- Comprender mejor los factores que influyen en el riesgo de diabetes.
- Mejorar las estrategias de prevención y políticas de salud pública para la diabetes.



### Descripción de las variables:

| **Nombre**               | **Descripción**                                                                                                                                          |
|--------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Diabetes_binary**       | 0 = no diabetes<br>1 = prediabetes<br>2 = diabetes                                                                                                      |
| **HighBP**                | 0 = no hipertensión<br>1 = hipertensión                                                                                                                 |
| **HighChol**              | 0 = no colesterol alto<br>1 = colesterol alto                                                                                                           |
| **CholCheck**             | 0 = no se ha realizado un chequeo de colesterol en 5 años<br>1 = sí se ha realizado un chequeo de colesterol en 5 años                                  |
| **BMI**                   | Índice de Masa Corporal (IMC)                                                                                                                            |
| **Smoker**                | ¿Has fumado al menos 100 cigarrillos en toda tu vida? [Nota: 5 cajetillas = 100 cigarrillos]<br>0 = no<br>1 = sí                                        |
| **Stroke**                | ¿Alguna vez te han dicho que tuviste un derrame cerebral?<br>0 = no<br>1 = sí                                                                            |
| **HeartDiseaseorAttack**  | Enfermedad coronaria (CHD) o infarto de miocardio (IM)<br>0 = no<br>1 = sí                                                                              |
| **PhysActivity**          | Actividad física en los últimos 30 días - excluyendo el trabajo<br>0 = no<br>1 = sí                                                                      |
| **Fruits**                | Consumes fruta 1 o más veces al día<br>0 = no<br>1 = sí                                                                                                  |
| **Veggies**               | Consumes vegetales 1 o más veces al día<br>0 = no<br>1 = sí                                                                                              |
| **HvyAlcoholConsump**     | (Hombres adultos >=14 bebidas por semana y mujeres adultas >=7 bebidas por semana)<br>0 = no<br>1 = sí                                                  |
| **AnyHealthcare**         | ¿Tienes algún tipo de cobertura de salud, incluyendo seguro médico, planes prepagos como HMO, etc.?<br>0 = no<br>1 = sí                                |
| **NoDocbcCost**           | ¿Hubo un momento en los últimos 12 meses en que necesitabas ver a un médico pero no pudiste por el costo?<br>0 = no<br>1 = sí                           |
| **GenHlth**               | ¿Cómo dirías que es tu salud en general?<br>Escala 1-5<br>1 = excelente<br>2 = muy buena<br>3 = buena<br>4 = regular<br>5 = mala                       |
| **MentHlth**              | Días de mala salud mental<br>Escala de 1 a 30 días                                                                                                       |
| **PhysHlth**              | Días de enfermedad física o lesión en los últimos 30 días<br>Escala de 1 a 30                                                                            |
| **DiffWalk**              | ¿Tienes serias dificultades para caminar o subir escaleras?<br>0 = no<br>1 = sí                                                                         |
| **Sex**                   | 0 = femenino<br>1 = masculino                                                                                                                            |
| **Age**                   | Categoría de edad en 13 niveles<br>1 = 18-24 años<br>2 = 25-29 años<br>3 = 30-34 años<br>4 = 35-39 años<br>5 = 40-44 años<br>6 = 45-49 años<br>7 = 50-54 años<br>8 = 55-59 años<br>9 = 60-64 años<br>10 = 65-69 años<br>11 = 70-74 años<br>12 = 75-79 años<br>13 = 80 o más años |                                                        |
| **Education**             | Nivel de educación<br>Escala de 1 a 6<br>1: No se graduó de la escuela secundaria<br>2: Graduado de la escuela secundaria<br>3: Asistió a la universidad<br>4: Graduado universitario<br>5: Estudios de posgrado<br>6: Doctorado o equivalente |
| **Income**                | Escala de ingresos<br>Escala de 1 a 8<br>1: <$10 K<br>2: $10–$15 K<br>3: $15–$20 K<br>4: $20–$25 K<br>5: $25–$35 K<br>6: $35–$50 K<br>7: $50–$75 K<br>8: >$75 K |


# Transformación y limpieza de los datos

In [1]:
#Importación de librerías
import pandas as pd
import numpy as np
import matplotlib.pyplot as t
import seaborn as sns

In [30]:
df['Age'].value_counts()

Age
10    10856
9     10112
8      8603
11     8044
7      6872
13     5426
12     5394
6      4648
5      3520
4      2793
3      2049
2      1396
1       979
Name: count, dtype: int64

In [16]:
#Importación de df

df = pd.read_csv('C:/Users/adolf/OneDrive/Documentos/diabetes_2.csv')
df.head()

Diabetes_binary  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0              0.0     1.0       0.0        1.0  26.0     0.0     0.0   
1              0.0     1.0       1.0        1.0  26.0     1.0     1.0   
2              0.0     0.0       0.0        1.0  26.0     0.0     0.0   
3              0.0     1.0       1.0        1.0  28.0     1.0     0.0   
4              0.0     0.0       0.0        1.0  29.0     1.0     0.0   

   HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                   0.0           1.0     0.0  ...            1.0   
1                   0.0           0.0     1.0  ...            1.0   
2                   0.0           1.0     1.0  ...            1.0   
3                   0.0           1.0     1.0  ...            1.0   
4                   0.0           1.0     1.0  ...            1.0   

   NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  Education  \
0          0.0      3.0       5.0      30.0       0.0  1.0   4.0        6.0   
1          0.0      3.0       0.0       0.0       0.0  1.0  12.0        6.0   
2          0.0      1.0       0.0      10.0       0.0  1.0  13.0        6.0   
3          0.0      3.0       0.0       3.0       0.0  1.0  11.0        6.0   
4          0.0      2.0       0.0       0.0       0.0  0.0   8.0        5.0   

   Income  
0     8.0  
1     8.0  
2     8.0  
3     8.0  
4     8.0  

[5 rows x 22 columns]

In [17]:
# Información de la estructura del df
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70692 entries, 0 to 70691
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Diabetes_binary       70692 non-null  float64
 1   HighBP                70692 non-null  float64
 2   HighChol              70692 non-null  float64
 3   CholCheck             70692 non-null  float64
 4   BMI                   70692 non-null  float64
 5   Smoker                70692 non-null  float64
 6   Stroke                70692 non-null  float64
 7   HeartDiseaseorAttack  70692 non-null  float64
 8   PhysActivity          70692 non-null  float64
 9   Fruits                70692 non-null  float64
 10  Veggies               70692 non-null  float64
 11  HvyAlcoholConsump     70692 non-null  float64
 12  AnyHealthcare         70692 non-null  float64
 13  NoDocbcCost           70692 non-null  float64
 14  GenHlth               70692 non-null  float64
 15  MentHlth           

In [18]:
# Información de los Null
df.isnull().sum()

Diabetes_binary         0
HighBP                  0
HighChol                0
CholCheck               0
BMI                     0
Smoker                  0
Stroke                  0
HeartDiseaseorAttack    0
PhysActivity            0
Fruits                  0
Veggies                 0
HvyAlcoholConsump       0
AnyHealthcare           0
NoDocbcCost             0
GenHlth                 0
MentHlth                0
PhysHlth                0
DiffWalk                0
Sex                     0
Age                     0
Education               0
Income                  0
dtype: int64

In [22]:
# Transformando los datos a entero (ya que ningún dato contiene números decimales > 0)
df = df.astype('int')

In [51]:
# Creando las funciones para crear columnas categóricas que ayuden con la lectura de las gráficas y sea más fácil para los usuarios

def status(x):
    if x['Diabetes_binary'] == 0:
        return "No Diabetes"
    else:
        return "Diabetes"
# ----------------------------------------------------

def hip(x):
    if x['HighBP'] == 0:
        return "No Hipertenso"
    else:
        return "Hipertenso"
# ----------------------------------------------------

def col(x):
    if x['HighChol'] == 0:
        return "No Colesterol alto"
    else:
        return "Colesterol alto"
# ----------------------------------------------------

def Sm(x):
    if x['Smoker'] == 0:
        return "No Fumador"
    else:
        return "Fumador"
# ----------------------------------------------------

def Sx(x):
    if x['Sex'] == 0:
        return "Femenino"
    else:
        return "Masculino"
# ----------------------------------------------------

def af(x):
    if x['PhysActivity'] == 0:
        return "Actividad física inactiva"
    else:
        return "Actividad física activa"
# ----------------------------------------------------

def fr(x):
    if x['Fruits'] == 0:
        return "No consume frutas"
    else:
        return "Consume frutas"
# ----------------------------------------------------

def vg(x):
    if x['Veggies'] == 0:
        return "No consume vegetales"
    else:
        return "Consume vegetales"
# ----------------------------------------------------

def al(x):
    if x['AnyHealthcare'] == 0:
        return "No tiene atención médica"
    else:
        return "Tiene atención médica"
# ----------------------------------------------------

def education(x):
    if x['Education'] == 1:
        return "No se graduó de la escuela secundaria"
    elif x['Education'] == 2:
        return "Graduado de la escuela secundaria"
    elif x['Education'] == 3:
        return "Asistió a la universidad"
    elif x['Education'] == 4:
        return "Graduado universitario"
    elif x['Education'] == 5:
        return "Estudios de posgrado"
    elif x['Education'] == 6:
        return "Doctorado o equivalente"
# ----------------------------------------------------

def age(x):
    if x['Age'] == 1:
        return "18-24 años"
    elif x['Age'] == 2:
        return "25-29 años"
    elif x['Age'] == 3:
        return "30-34 años"
    elif x['Age'] == 4:
        return "35-39 años"
    elif x['Age'] == 5:
        return "40-44 años"
    elif x['Age'] == 6:
        return "45-49 años"
    elif x['Age'] == 7:
        return "50-54 años"
    elif x['Age'] == 8:
        return "55-59 años"
    elif x['Age'] == 9:
        return "60-64 años"
    elif x['Age'] == 10:
        return "65-69 años"
    elif x['Age'] == 11:
        return "70-74 años"
    elif x['Age'] == 12:
        return "75-79 años"
    elif x['Age'] == 13:
        return "80 o más años"


In [42]:
# Creando las variables categóricas

df['Status'] = df.apply(status, axis = 1)

df['Hipertenso'] = df.apply(hip, axis = 1)

df['Nivel_Col'] = df.apply(col, axis = 1)

df['Género'] = df.apply(Sx, axis = 1)

df['Actividad_física'] = df.apply(af, axis = 1)

df['Consumo_Frutas'] = df.apply(af, axis = 1)

df['Consumo_vegetales'] = df.apply(fr, axis = 1)

df['Seguro_med'] = df.apply(al, axis = 1)

df['Educación'] = df.apply(education, axis = 1)

df['Edad'] = df.apply(age, axis = 1)

In [46]:
# Visualizando las nuevas variables
df[['Status', 'Hipertenso', 'Nivel_Col', 'Género', 'Actividad_física',
       'Consumo_Frutas', 'Consumo_vegetales', 'Seguro_med', 'Educacion',
       'Edad', 'Educación']].head()

Status     Hipertenso           Nivel_Col     Género  \
0  No Diabetes     Hipertenso  No Colesterol alto  Masculino   
1  No Diabetes     Hipertenso     Colesterol alto  Masculino   
2  No Diabetes  No Hipertenso  No Colesterol alto  Masculino   
3  No Diabetes     Hipertenso     Colesterol alto  Masculino   
4  No Diabetes  No Hipertenso  No Colesterol alto   Femenino   

            Actividad_física             Consumo_Frutas  Consumo_vegetales  \
0    Actividad física activa    Actividad física activa  No consume frutas   
1  Actividad física inactiva  Actividad física inactiva     Consume frutas   
2    Actividad física activa    Actividad física activa     Consume frutas   
3    Actividad física activa    Actividad física activa     Consume frutas   
4    Actividad física activa    Actividad física activa     Consume frutas   

              Seguro_med                Educacion           Edad  \
0  Tiene atención médica  Doctorado o equivalente     35-39 años   
1  Tiene atención médica  Doctorado o equivalente     75-79 años   
2  Tiene atención médica  Doctorado o equivalente  80 o más años   
3  Tiene atención médica  Doctorado o equivalente     70-74 años   
4  Tiene atención médica     Estudios de posgrado     55-59 años   

                 Educación  
0  Doctorado o equivalente  
1  Doctorado o equivalente  
2  Doctorado o equivalente  
3  Doctorado o equivalente  
4     Estudios de posgrado

# Análisis de los datos (Estadísticos y visuales )

In [78]:
#Análisis estadístico de las variables que nos pueden aportar información valiosa para las hipótesis (Dado que la mayoría contienen solo 0 y 1)
pd.options.display.float_format = "{:,.2f}".format
df[['BMI','GenHlth','MentHlth','PhysHlth','Age','Education','Income']].describe()

BMI   GenHlth  MentHlth  PhysHlth       Age  Education    Income
count 70,692.00 70,692.00 70,692.00 70,692.00 70,692.00  70,692.00 70,692.00
mean      29.86      2.84      3.75      5.81      8.58       4.92      5.70
std        7.11      1.11      8.16     10.06      2.85       1.03      2.18
min       12.00      1.00      0.00      0.00      1.00       1.00      1.00
25%       25.00      2.00      0.00      0.00      7.00       4.00      4.00
50%       29.00      3.00      0.00      0.00      9.00       5.00      6.00
75%       33.00      4.00      2.00      6.00     11.00       6.00      8.00
max       98.00      5.00     30.00     30.00     13.00       6.00      8.00

### Interpretación de Resultados

1. **BMI (Índice de Masa Corporal):**
   - **Media:** 29.86 - Indica sobrepeso, ya que un IMC de 25-29.9 se considera sobrepeso y 30+ es obesidad.
   - **Rango:** 12-98 - Gran variabilidad en los IMC, desde bajo peso hasta obesidad severa.

2. **GenHlth (Salud General):**
   - **Media:** 2.84 - Salud general promedio entre "buena" y "regular", en una escala de 1 (excelente) a 5 (pobre).
   - **Distribución:** Mayoría con percepción de salud entre "buena" y "regular".

3. **MentHlth (Salud Mental - días de mala salud mental):**
   - **Mediana:** 0 - La mayoría no reporta días de mala salud mental, aunque hay casos severos con hasta 30 días.

4. **PhysHlth (Salud Física - días de mala salud física):**
   - **Mediana:** 0 - Similar a la salud mental, la mayoría no reporta problemas, pero algunos tienen problemas físicos graves.

5. **Age (Edad):**
   - **Media:** 8.58 - Promedio en la categoría de 55-59 años, indicando una muestra mayor.
   - **Distribución:** Bien distribuida, pero sesgada hacia una población mayor.

6. **Education (Educación):**
   - **Media:** 4.92 - La mayoría tiene educación superior, con muchos completando la universidad o estudios de posgrado.

7. **Income (Ingresos):**
   - **Media:** 5.70 - Ingreso promedio entre $35,000 y $50,000 anuales, indicando una clase media predominante.

La muestra tiene un sobrepeso significativo, está orientada hacia una población mayor con buena educación y un ingreso de clase media. Hay poca prevalencia de problemas de salud mental y física reportados, aunque hay casos extremos en ambos.
